src/Session.cpp

```diff
@@ -92,14 +93,32 @@ Session::Session(const Session &s)
     clean();
 }
 
+void Session::deepCopyUsers(const std::unordered_map<std::string, User*>& newUsers)
+{
```

Comment:
as I understand, you don't "clean" the original map
so it will have both members...

Reply:
Well yeah, it's a copy. I am not supposed to clean someone else's memory in this function